Unused Stapler Facet is removed

jgreffe opened the pull request:
Fixes #87 

Remove all classes related to Facet
Remove all plugin.xml declarations related to Facet
Fixes infinite loop in `JexlInspection.tokenize("'")` : case `${%Test('`
Fixes out of range issue with patterns like `${%Test(something)}` (`Argument rangeInElement (20,21) endOffset must not exceed descriptor text range (121, 141) length (20)`)

Upgrading to this new plugin version on existing projects having facet will popup a warning message:
![Monosnap Error 2022-11-02 12-50-04](https://user-images.githubusercontent.com/36693362/199484083-6148e13c-0bfc-4e2c-82d6-a585e8300443.png)


- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

<!--
Put an `x` into the [ ] to show you have filled the information.
The template comes from https://github.com/jenkinsci/.github/blob/master/.github/pull_request_template.md 
You can override it by creating .github/pull_request_template.md  in your own repository 
-->

duemir commented:
@timja I am on the fence about this. What do you think?

timja commented:
Sounds fine to remove if it’s not being used, but not too familiar with this

duemir commented:
OK. We should probably do it. It is unlikely that this Facet was ever used in the field by anybody but me. 
But I think we should add a few more tests for JexlInspection, which is being published to a wider audience by this change. e.g., test that it handles property lookups (l10n) like `${%Name}` and `${%Build Queue(items.size())}`

jgreffe commented:
Added some unit tests, and fixed two issues:
- infinite loop in JexlInspection.tokenize("'")
- out of range issue with patterns like `${%Test(something)}` (Argument rangeInElement (20,21) endOffset must not exceed descriptor text range (121, 141) length (20))

duemir commented on src/test/java/org/kohsuke/stapler/idea/JexlInspectionTest.java:
```diff
@@ -43,4 +35,48 @@ public void testParserFailureJexlInspection() {
+                         jexlInspection.tokenize(expectation.getKey()));
+        }
+    }
+};
```
```suggestion
}
```

duemir commented on src/test/java/org/kohsuke/stapler/idea/JexlInspectionTest.java:
```diff
@@ -43,4 +35,48 @@ public void testParserFailureJexlInspection() {
+    }
+
+    public void testTokenize() {
+        Map<String, String> expectations = new HashMap<String, String>() {
```
```suggestion
        Map<String, String> expectations = new HashMap<>() {
```

duemir commented on src/test/java/org/kohsuke/stapler/idea/JexlInspectionTest.java:
```diff
@@ -11,24 +11,16 @@
 
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
```
imports need to be cleaned up.

duemir reviewed (changes requested):
A few minor suggestions. 

I've confirmed the test fail and that changes in Jexl Inspection fix them. I have a hard time reasoning about this thing, so this is as best I can do in a limited time.

duemir commented:
Oh boy! I built a local version of the next release. Ran the JEXL inspection on @jenkinsci/jenkins . It crashes in two files and shows 11 errors in 9 files. 

4 has to do with localization. The rest is about HTML-encoded angle bracket 🤔

jgreffe commented:
@duemir : this means as jexl inspection is now activated by default, it displays proper errors, but also point out on faulty plugin code (crashes) ?

duemir commented:
@jgreffe Yes, as far as I can see. I will try to iron at least some out.

jgreffe commented:
Crashes occur with i18n like:
```
<j:jelly xmlns:j="jelly:core">
    <p>
        ${%Te'st}</p>
</j:jelly>
```
